tests/test_dupefilters.py

```diff
@@ -2,22 +2,25 @@
 import shutil
 import sys
 import tempfile
-import unittest
 from pathlib import Path
 
 from testfixtures import LogCapture
+from twisted.internet.defer import inlineCallbacks
+from twisted.trial import unittest
 
 from scrapy.core.scheduler import Scheduler
 from scrapy.dupefilters import RFPDupeFilter
 from scrapy.http import Request
 from scrapy.utils.python import to_bytes
 from scrapy.utils.test import get_crawler
-from tests.spiders import SimpleSpider
+from tests.spiders import NoRequestsSpider
 
 
+@inlineCallbacks
 def _get_dupefilter(*, crawler=None, settings=None, open=True):
```

Comment:
Same as with `DownloaderMiddlewareManager`, it's created via the engine, so it would make sense if we run `crawler.crawl()` and doesn't if we don't, except that here we also open and close the dupefilter explicitly, which can be problematic if it was already opened and closed during `crawler.crawl()`.